Weather/Views/WeatherViewController.swift

```diff
@@ -55,15 +54,21 @@ extension WeatherViewController: UICollectionViewDataSource, UICollectionViewDel
         if collectionView == view.timedForecastView {
             let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "timedForecastCell", for: indexPath) as! TimedForecastCellView
             
-            timedForecastPresenter!.activateTimeLabelConstraints(view: cell.timeLabel)
-            timedForecastPresenter!.activateTemperatureLabelConstraints(view: cell.temperatureLabel)
-            
-            let item = timedForecastPresenter!.getItem(at: indexPath.row)
+            let item = timedForecastPresenter?.getItem(at: indexPath.row) ?? TimedForecast(at: "Day", is: 0)
             
-            cell.configure(time: item.getTime(), temperature: item.getTemperature())
+            cell.display(time: item.getTime(), temperature: item.getTemperature())
 
             return cell
         }
+        if collectionView == view.weekForecastView {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "weekForecastCell", for: indexPath) as! WeekForecastCellView
+            
+            let item = weekForecastPresenter?.getItem(at: indexPath.row) ?? WeekdayForecast(on: "Day", temperatureAtMidday: 0, temperatureAtNight: 0)
```

Comment:
I dont think that making empty cell in this case is good approach as it is clearly some issue if you dont get an item for row. For the presenters you can make them implicitly unwrapped in declaration to avoid optional chaining.

Reply:
> For the presenters you can make them implicitly unwrapped in declaration to avoid optional chaining.

I guessed it's bad practice to use unwrapping at declaration, or in such cases it's ok?

Reply:
@declipz  Yes, this is so called "early crash" principle. You should spot such programmer mistakes while developing. If the cell is not dequeued for some reason – this is a programmer mistake, not the application logic.